xhttp/request.go

```diff
@@ -60,3 +62,13 @@ func NewRequestWithContext(ctx context.Context, method, url string, body io.Read
 	req.Header.Set("User-Agent", defaultUserAgent)
 	return req, nil
 }
+
+// NewJSONRequest is a wrapper that will marshall the given data as a JSON and call [NewRequestWithContext].
+// It makes it practical to create HTTP requests that send JSON bodies with detailed user agents (what we want most of the time).
+func NewJSONRequest(ctx context.Context, method, url string, data any) (*http.Request, error) {
```

Comment:
Adding WithContext seemed like too much here, in general we always pass a context to ensure cancellation/timeouts.

Reply:
I thought the same :heart: